src/overlays/actors/ovl_En_Jc_Mato/z_en_jc_mato.c

```diff
@@ -19,20 +28,111 @@ const ActorInit En_Jc_Mato_InitVars = {
     (ActorFunc)EnJcMato_Init,
     (ActorFunc)EnJcMato_Destroy,
     (ActorFunc)EnJcMato_Update,
-    (ActorFunc)EnJcMato_Draw
+    (ActorFunc)EnJcMato_Draw,
+};
+
+// was D_80B9E210
+ColliderSphereInit sSphereInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_SPHERE,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    { 0, { { 0, 0, 0 }, 15 }, 100 },
+};
+
+DamageTable EnJcMatoDamageTable = {
+    0x01, 0x01, 0x01, 0x01, 0x01, 0xF1, 0x01, 0x01, 0x01, 0x01, 0x01, 0xF1, 0xF1, 0xF1, 0x01, 0x01,
+    0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Jc_Mato_0x80B9DEE0/func_80B9DEE0.asm")
+s32 func_80B9DEE0(EnJcMato* this, GlobalContext* globalCtx) {
+    this->collider.dim.worldSphere.center.x = this->JcMatoPos.x;
+    this->collider.dim.worldSphere.center.y = this->JcMatoPos.y;
+    this->collider.dim.worldSphere.center.z = this->JcMatoPos.z;
+    if ((this->collider.base.acFlags & 2) && !this->hitFlag && (this->actor.colChkInfo.damageEffect == 0xF)) {
+        this->collider.base.acFlags &= 0xFFFD;
+        Audio_PlayActorSound2(&this->actor, 0x4807);
+        globalCtx->interfaceCtx.unk25C = 1;
+        this->hitFlag = 1;
+        return 1;
+    } else {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+        return 0;
+    }
+}
+
+void func_80B9DFC8(EnJcMato* this) {
+    this->actionFunc = func_80B9DFDC;
+}
+
+void func_80B9DFDC(EnJcMato* this, GlobalContext* globalCtx) {
+    s16 phi_v1;
```

Comment:
This looks like a flag to track if unk1A0 has reached 0 or not yet, so I think a better name for this temp is probably possible. To me it looks like the target spawns with a timer (unk1A0) of 25, then once it gets hit, it counts down to 0, then it despawns. So unk1A0 would be a `despawnTimer` (or something), and phi_v1 would just be something that reflects that. That being said, I don't see the purpose of phi_v1 here. Not sure why they didn't just check for the value of unk1A0.

Reply:
I had thought about it being a despawn timer, so I opened KZ and did the whole boat ride without hitting a target, but I never saw it despawn....could just be instantaneous though? I do agree that it looks to be a counter of some kind. I tried removing the phi temp, but it wouldn't match, I could try re-arranging things though.

Reply:
I think the timer doesn't start counting down until you hit the target. When you hit a target, it kinda flies backward for a bit and then despawns, so I'm guessing that after you hit it, it counts from 25 down to 0 before it disappears

Reply:
I did not think about that! you could be right and the target despawns after 25 frames or something. I'll check :D